moshi/src/main/java/com/squareup/moshi/StandardJsonAdapters.java

```diff
@@ -215,6 +223,31 @@ static int rangeCheckNextInt(JsonReader reader, String typeMessage, int min, int
     }
   };
 
+  /**
+   * Loads the generated JsonAdapter for classes annotated {@link JsonClass}. This works because it
+   * uses the same naming conventions as {@code JsonClassCodeGenProcessor}.
+   */
+  static JsonAdapter<?> generatedAdapter(Moshi moshi, Type type, Class<?> rawType) {
+    String adapterClassName = rawType.getName().replace("$", "_") + "JsonAdapter";
+    try {
+      @SuppressWarnings("unchecked") // We generate types to match.
+      Class<? extends JsonAdapter<?>> adapterClass = (Class<? extends JsonAdapter<?>>)
+          Class.forName(adapterClassName, true, rawType.getClassLoader());
+      if (type instanceof ParameterizedType) {
+        return adapterClass.getDeclaredConstructor(Moshi.class, Type[].class)
+            .newInstance(moshi, ((ParameterizedType) type).getActualTypeArguments());
+      } else {
+        return adapterClass.getDeclaredConstructor(Moshi.class).newInstance(moshi);
+      }
+    } catch (ClassNotFoundException | NoSuchMethodException e) {
+      throw new RuntimeException(
+          "Failed to find the generated the JsonAdapter for " + rawType, e);
+    } catch (IllegalAccessException | InvocationTargetException | InstantiationException e) {
```

Comment:
Doesn't this widen to ReflectiveOperationException which is only available on API 19+ / Java 7+?

Reply:
The compiler can’t do that automatically because it doesn’t want to catch other, unknown subtypes of ReflectiveOperationException.

Reply:
That contrasts with AS warnings and animal sniffer failures that I've received in the past.

Reply:
Confirmed you are correct. Gross! Fixed to have 5 catch clauses. Gross!